src/checkL2Version.ts

```diff
@@ -0,0 +1,73 @@
+import { exec } from "child_process";
+import * as vscode from "vscode";
+import { getShowLama2Term } from "./utils";
+
+const MIN_VERSION_TO_CHECK = "1.5.2";
+const LAMA2_TERM_NAME = "AutoLama2";
+const UPDATE_MSG = "Support for environment variables.";
+
+export function checkL2Version() {
+  try {
+    getL2Version((l2Version) => {
+      // Check if version is below MIN_VERSION_TO_CHECK
+      if (compareL2Versions(l2Version, MIN_VERSION_TO_CHECK) < 0) {
+        showUpdateWarning();
+      }
+    });
+  } catch (e: any) {
+    console.log("Problem while checking for version -> ", e);
+  }
+}
+
+function getL2Version(callback: (version: string) => void) {
+  exec("l2 --version", (error, stdout, stderr) => {
+    const versionOutput = stdout.trim();
+    const versionMatch = versionOutput.match(/v(\d+\.\d+\.\d+)/);
+    if (versionMatch && versionMatch.length === 2) {
```

Comment:
Why `length === 2`?  We can use capture group to get a triple for (major, minor, patch) as well directly, I think.

Reply:
I think we can try to use the official semver regex. See [here](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string) for details. Check the actual regex in action at [regex101](https://regex101.com/r/vkijKf/1/)

Reply:
Damn never knew so many kinds are there